### DISCUSSION TITLE
We need zip js for Angular 11

### DISCUSSION THREAD

**Discussion by DJ-991** [gildas-lormeau/zip.js#476] [Q&A]
How can we use zip.js in Angular11? I need to encrypt zip having pdf. Please suggest, after npm install how can we use zip.js in angular framework? We have blob for the PDF.

**Comment by gildas-lormeau** [comment]
What should be particular to Angular 11?

**Comment by DJ-991** [comment]
Just in angular how can we implement zip.js

**Comment by gildas-lormeau** [comment]
Like in any other JavaScript/TypeScript project?

**Comment by DJ-991** [comment]
This is angular project with typescript, we need to encrypt the zip having pdf blob. Can we do this with zip.js? 

Ask- whenever user open that downloaded zip, it should ask for the password.

**Comment by gildas-lormeau** [comment]
Yes, it is probably possible to implement this. What's blocking you?

**Comment by DJ-991** [comment]
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/30df3599-d489-4a3b-872b-b38db1926de4)

is this code is fine for encryption and zip that pdf blob? blob we are getting from API

**Comment by DJ-991** [comment]
PFA  - by running:
$ npm install @zip.js/zip.js

and importing the libary as an ECMAScript module:
import * as zip from "@zip.js/zip.js";

using like this : useCompressionStream is giving error in code 
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/1b350b37-b4c7-4740-960f-65adc612ea47)

**Comment by gildas-lormeau** [comment]
`useCompressionStream` is an option passed to `zip.configure()`. See here for an example of code using encryption: https://github.com/gildas-lormeau/zip.js/blob/master/tests/all/test-crypto.js.

**Comment by DJ-991** [comment]
https://github.com/gildas-lormeau/zip.js/blob/master/tests/all/test-crypto.js
should we use that code(test-crypto.js) for encryption in our typescript code and for zipping ? @gildas-lormeau

**Comment by gildas-lormeau** [comment]
This is a code example that shows how to write a zip file with an AES password so I assume it's similar to what you want to do.

**Comment by DJ-991** [comment]
Will try this code hope this will work. @gildas-lormeau

**Comment by gildas-lormeau** [comment]
You'll probably have to follow these instructions as well: https://github.com/gildas-lormeau/zip.js/discussions/229

**Comment by DJ-991** [comment]
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/344d1b3a-1947-4fc1-ae64-2545de0add2a)
getting this error after config

here it is:
1: create file inside src - 
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/8760b951-2dfb-4da7-9cf8-9006d2783e40)

2: angular.json "serve" config
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/7804245f-6678-4f91-a345-42fc59f29213)

3: angular.json "builder" and "options"
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/75566d52-7f73-47c5-9de9-67b7bc80a57b)

4: package.json
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/b4676d83-6e24-4bf0-b970-c4e8d03565ff)


My angular version is Angular 11. so i installed npm install -D @angular-builders/custom-webpack@11.1.1

**Comment by gildas-lormeau** [comment]
It looks like the ES version in your project is too old. Here is the fix of a similar issue in Angular CLI:
 - https://github.com/angular/angular-cli/commit/dd60228974a75b3c6b2ae98521444f0354161240
 - https://github.com/angular/angular-cli/issues/21196

**Comment by DJ-991** [comment]
in which file we need to change tsconfig.json or package.json ? not getting 
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/a24049db-9d82-4264-8e05-abcb23e457ee)

**Comment by gildas-lormeau** [comment]
I'm sorry but I don't know because I don't use Angular.

**Reply by DJ-991** [reply]
is there any cdn available for zip.js ?

**Comment by DJ-991** [comment]
is there any cdn available for zip.js ? @gildas-lormeau

**Comment by DJ-991** [comment]
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/232265fd-ba62-4335-ba97-3187b18985e1)
this is the error coming. in catch block after passing the blob application.pdf

**Reply by gildas-lormeau** [reply]
This error is normal, it's a test to verify everything works properly (i.e. missing and invalid password). The code which does not throw an error is at line 497 in your code.

**Reply by DJ-991** [reply]
i tried that test-crypto.js code, having text blob, its not giving encrypted zip file in data variable, its just giving the text file without encryption. In line no 511 its downloading... as per above screenshot. @gildas-lormeau

**Reply by DJ-991** [reply]
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/570ff630-95c6-4e9a-8aa1-e5454780bd22)
